File: Modules/SEDO/EMS/SIE.Web.EMS/FixedAssets/Accounts/Commands/WithdrawCommand.js
SIE.defineCommand('SIE.Web.EMS.FixedAssets.Accounts.Commands.WithdrawCommand', {
    meta: { text: "撤回", group: "edit", iconCls: "icon-FileReturn icon-blue" },
    canExecute: function (view) {
        var selectModels = view.getSelection();
        if (selectModels.length != 1) return false;
        var res = true;
        SIE.each(selectModels, function (model) {
            if (model.data.ReviewStatus !== 20) {
                res = false;
                return false;
            }
        });
        return res;
    },

    execute: function (view) {
        var modelData = view.getCurrent().data;
        view.execute({
            data: modelData,
            success: function (res) {
                SIE.Msg.showMessage("撤回成功!".t());
                view.reloadData();
            }
        });
    }
});

File: Modules/SEDO/EMS/SIE.Web.EMS/FixedAssets/Accounts/Commands/WithdrawCommand.cs
using SIE.EMS.FixedAssets.Accounts;
using SIE.Web.Command;

namespace SIE.Web.EMS.FixedAssets.Accounts.Commands
{
    /// <summary>
    /// 撤回
    /// </summary>
    [JsCommand("SIE.Web.EMS.FixedAssets.Accounts.Commands.WithdrawCommand")]
    public class WithdrawCommand : ViewCommand
    {
        /// <summary>
        /// 撤回保存
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scop</param>
        /// <returns>bool</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var fixedAssetsAccount = args.Data.ToJsonObject<FixedAssetsAccount>();
            return RT.Service.Resolve<FixedAssetsAccountController>().Withdraw(fixedAssetsAccount.Id);
        }
    }
}
